satellite/satellitedb/overlaycache.go

```diff
@@ -899,7 +916,15 @@ func (cache *overlaycache) UpdateUptime(ctx context.Context, nodeID storj.NodeID
 		return nil, Error.Wrap(errs.Combine(errs.New("unable to get node by ID: %v", nodeID), tx.Rollback()))
 	}
 
-	return getNodeStats(dbNode), Error.Wrap(tx.Commit())
+	err = tx.Commit()
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	stats = getNodeStats(dbNode)
+	cache.updateChecker.SetNodeStats(nodeID, isUp, stats)
```

Comment:
What if this races with some other update uptime?

Reply:
It is possible to race, but I left it because the original code is also susceptible to race conditions. That is, if two `UpdateUptime` calls are made from two different routines, with different information, there's no guarantee which will be applied last (and persisted) in the db. Because of that, it seemed better to keep the solution simpler and allow for a short-term race issue.

If you feel the potential race is unacceptable, let me know and I'll happily change it, though it will complicate the solution.

Reply:
Yeah, but they have a tx boundary to serialize the changes. This could end up caching a DQ-ed node in the updateChecker as being valid. But, yeah, I'm not sure how much damage it would do.

Reply:
It's true, but only for the duration specified, if that makes sense. If we hold the cache for each node for a limited time, it reduces the risk of being out-of-date in a way that would actually affect the system ... I think. :)

I'd appreciate any more thoughts on this, I did give it some thought while developing but thought that the risk of problems was low enough that it didn't justify the added complexity. I'll give it more thought, see if I can think of a different way to do it.

Reply:
I'm trying to figure out the potential problem cases... so here's the table:

```
db   time        1s     30s   1m
--------------------------------
up   no-cache    up!    up    up
up   no-cache    do!    up    up
do   no-cache    do!    up    up
do   no-cache    up!    up    up

up   cached      up     up    up!
up   cached      do!    up!   up
do   cached      do     up!   up
do   cached      up!    up!   up
```
up - up, do - down, ! - flush to database.

The only concerning case seems to be:
```
up   cached      up     up    up!
up   cached      up!    up!   up!
```

It seems to be that the result of these will end up with a different uptime count.